src/networking.c

```diff
@@ -2240,6 +2248,42 @@ int clientSetNameOrReply(client *c, robj *name) {
     return C_OK;
 }
 
+/* Reset the client state to resemble a newly connected client.
+ */
+void resetCommand(client *c) {
```

Comment:
This drops the CLIENT_UNIX_SOCKET flag, which should probably be retained.

Reply:
@madolson Good catch, thanks! It's actually more than just `CLIENT_UNIX_SOCKET` that should be retained, so it looks safer to only clear specific flags.

Reply:
New approach looks better.